preprocessing/json_preprocessor.py

```diff
@@ -67,7 +67,7 @@ def extract_json_objects(raw_string_data: str, start_json_object: list,
     current_position = 0
 
     for start_i in start_json_object:
```

Comment:
Question: Is the `start_json_object` supposed to have integer values? Does the `start_i` variable refer to the index of an item in the collection or the item itself? If you're trying to go over the collection by indexes, this will break and you will need to use `enumerate` instead.